$OpenBSD: patch-configure_ac,v 1.12 2020/11/25 14:26:26 sthen Exp $

- libressl pkg-config file claims to be 1.0.0
- libdl is not used on OpenBSD
- various tools print OPENSC_SCM_REVISION with --version; set it to
something sensible. (.git doesn't exist in the release tarballs so the
above test to check if it's a git repo can be ignored).

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -21,7 +21,7 @@ define([VS_FF_PRODUCT_URL], [https://github.com/OpenSC
 
 m4_sinclude(version.m4.ci)
 
-m4_define([openssl_minimum_version], [1.0.1])
+m4_define([openssl_minimum_version], [1.0.0])
 
 AC_INIT([PRODUCT_NAME],[PACKAGE_VERSION_MAJOR.PACKAGE_VERSION_MINOR.PACKAGE_VERSION_FIX[]PACKAGE_SUFFIX],[PRODUCT_BUGREPORT],[PRODUCT_TARNAME],[PRODUCT_URL])
 AC_CONFIG_AUX_DIR([.])
@@ -386,7 +386,7 @@ if test "${GIT_CHECKOUT}" = "yes"; then
 	OPENSC_SCM_REVISION="OpenSC-${REVISION_DESCRIPTION}, ${HASH_COMMIT_DATE}"
 	OPENSC_VERSION_REVISION="$(${GIT} rev-list ${GIT_TAG_COMMIT}..HEAD --count || echo 0)"
 else
-	OPENSC_SCM_REVISION="No Git revision info available"
+	OPENSC_SCM_REVISION="${PACKAGE_STRING}"
 	OPENSC_VERSION_REVISION="0"
 fi
 
@@ -454,11 +454,6 @@ AC_CHECK_LIB(
 )
 
 if test "${WIN32}" = "no"; then
-	dnl dl support
-	AC_SEARCH_LIBS([dlopen], [dl dld], [], [
-		AC_MSG_ERROR([unable to find the dlopen() function])
-	])
-
 	dnl Special check for pthread support.
 	AX_PTHREAD(
 		[AC_DEFINE(
